src/aero/fsi/FSIturbine.C

```diff
@@ -31,6 +32,68 @@ namespace sierra {
 
 namespace nalu {
 
+/** Project a point 'pt' onto a line from 'lStart' to 'lEnd' and return the
+   non-dimensional location of the projected point along the line in [0-1]
+   coordinates \f[ nonDimCoord = \frac{ (\vec{pt} - \vec{lStart}) \cdot (
+   \vec{lEnd} - \vec{lStart} ) }{ (\vec{lEnd} - \vec{lStart}) \cdot (\vec{lEnd}
+   - \vec{lStart}) } \f]
+*/
+KOKKOS_FUNCTION
+double projectPt2Line(
+  const Point& pt,
+  const Point& lStart,
+  const Point& lEnd)
+{
+
+  double nonDimCoord = 0.0;
+
+  double num = 0.0;
+  double denom = 0.0;
+
+  for (int i = 0; i < 3; i++) {
+    num += (pt[i] - lStart[i]) * (lEnd[i] - lStart[i]);
+    denom += (lEnd[i] - lStart[i]) * (lEnd[i] - lStart[i]);
+  }
```

Comment:
If we use `vs::Vectors` this whole loop reduces to:
```suggestion
num = (pt-lStart)&(lEnd-lStart);
dnom = vs::mag_sqr(lEnd-lStart);
```

Reply:
ok, I'll look into that. I hadn't noticed `vs::Vector`.

Reply:
Yeah it is new. We pulled it from AMR-Wind to help with some of the vector math in the fsi stuff. It is a device compatible data structure.